src/plugins/createPluginManager.ts

```diff
@@ -0,0 +1,28 @@
+import {Plugin, PluginSlots} from '@plugins/PluginsContext';
+
+export class PluginManager {
+  private plugins: Plugin[];
+
+  public constructor() {
+    this.plugins = [];
+  }
+
+  public add(plugin: Plugin): void {
+    plugin.order ? this.plugins.splice(plugin.order, 0, plugin) : this.plugins.push(plugin);
```

Comment:
Order isn't an index, it's a priority. Imagine in example:

```ts
existingOrders = [ 5, 5, 5, 3, 2, 1 ]
newOrder = 1
newOrders = [ 5, 1, 5, 5, 3, 2, 1 ] // wrong
```

Besides, it's good to have this array immutable, so the changes could be detected.

```suggestion
    const order = plugin.order ?? -Infinity;
    const index = this.plugins.findIndex(item => item.order < order);
    if (index === -1) {
      this.plugins = [...this.plugins, plugin];
    } else {
      this.plugins = [...this.plugins.slice(0, index), plugin, ...this.plugins.slice(index)];
    }
```